src/releasing/quattorpoddoc.py

```diff
@@ -13,278 +16,384 @@
-    output = run_asyncloop("mvn clean", startpath=modules_location)
-    LOGGER.debug(output)
-
-    LOGGER.info("Doing maven compile in %s." % modules_location)
-    output = run_asyncloop("mvn compile", startpath=modules_location)
-    LOGGER.debug(output)
+    logger.info("Doing maven compile in %s." % repoloc)
+    output = run_asyncloop("mvn compile", startpath=repoloc)
+    logger.debug(output)
 
 
-def generatemds(pods, location):
+def generatemds(repo, sources, location):
     """
     Takes a list of components with podfiles and generates a md file for it.
     """
-    LOGGER.info("Generating md files.")
-    counter = 0
+    logger.info("Generating md files.")
     mdfiles = []
 
-    comppath = os.path.join(location, DOCDIR, SUBDIR)
+    comppath = os.path.join(location, DOCDIR, REPOMAP[repo]['sitesubdir'])
     if not os.path.exists(comppath):
         os.makedirs(comppath)
 
-    for component in sorted(pods):
-        for pod in pods[component]:
-            component = component.replace('ncm-', '')
-            title = os.path.splitext(os.path.basename(pod))[0]
-            if component != title:
-                title = component+'::'+title
-            mdfile = os.path.join(comppath, '%s.md' % title)
-            convertpodtomarkdown(pod, mdfile)
-            counter += 1
+    for source in sources:
+        if source.endswith(".pan"):
+            mdfile = createmdfrompan(source, comppath)
             mdfiles.append(mdfile)
+        else:
+            sourcename = source.split(REPOMAP[repo]['target'])[-1]
+            mdfile = os.path.splitext(sourcename)[0].replace("/", "::").lower() + ".md"
 
-    LOGGER.info("Written %s md files." % counter)
+            convertpodtomarkdown(source, os.path.join(comppath, mdfile))
+            mdfiles.append(mdfile)
+
+    logger.info("Written %s md files." % len(mdfiles))
     return mdfiles
 
 
+def createmdfrompan(source, comppath):
+    """
+    Takes a pan schema, creates the pan annotations and parses them to markdown.
+    """
+    modname = os.path.basename(os.path.split(source)[0])
+    mdfile = modname + "::schema.md"
+    tmpdir = tempfile.mkdtemp()
+    logger.debug("Temporary directory: %s " % tmpdir)
+    panccommand = ["panc-annotations", "--output-dir", tmpdir, "--base-dir"]
+    panccommand.extend(os.path.split(source))
+    output = run_asyncloop(panccommand)
+    logger.debug(output)
+    namespace = "{http://quattor.org/pan/annotations}"
+
+    tpl = "schema.pan.annotation.xml"
+    xml = etree.parse(os.path.join(tmpdir, tpl))
+    root = xml.getroot()
+
+    with open(os.path.join(comppath, mdfile), "w") as fih:
+        fih.write("# Types\n\n")
+        for stype in root.findall('%stype' % namespace):
+            name = stype.get('name')
+            fih.write("- /software/%s/%s\n" % (modname, name))
+
+            for doc in stype.findall(".//%sdesc" % namespace):
+                fih.write("    - decription: %s\n" % doc.text)
```

Comment:
the indentation needs to be parameterised. (there's no way to review if the indentation is correct/identical). use e.g.
```python
txt.append("%s- description: %s"% (" "*4, doc.text))
```
for 4 space indentation